daemon/list.go

```diff
@@ -249,7 +249,7 @@ func (daemon *Daemon) foldFilter(view container.View, config *types.ContainerLis
 	err := psFilters.WalkValues("exited", func(value string) error {
 		code, err := strconv.Atoi(value)
 		if err != nil {
-			return err
+			return filters.InvalidFilter{Filter: "exited", Value: value}
```

Comment:
If possible, I think we should avoid exporting `InvalidFilter`, as it's mostly a "convenience" function to format an `errdefs.InvalidParam()`. Basically the "contact" is to return an `errdef` type, and that it happens to be "because of an invalid filter" is mostly an internal things for the filters package.

We could either decide to have a local helper, or inline for these two occurrences (`return errdefs.InvalidParam(errors.Errorf("invalid filter ...")`)

Or we could consider having a method to get Integer values (`filters.GetInt(name, <default>)`), like we added for booleans. Of course the ideal would've been for the filter values to be typed, but I guess that requires more work.

Reply:
I like the `filters.GetInt` approach, but didn't want to bloat the filters API too much. This would allow us to completely unexport invalidFilter though. I'll give it a shot.